### ISSUE TITLE
java.time.format.DateTimeParseException

### ISSUE THREAD

**Issue by MasterShu** [aws-amplify/amplify-flutter#1131]
**Describe the bug**
I've met a bug when I update a model.
`Unresolved exception class when finding catch block: java.time.format.DateTimeParseException`

**To Reproduce**
1. I have a model that has a filed type is AWSDateTime
2. It could be creating a new model data correctly
3. When I update this model data
4. I've got this error

```
W/art     (24813): Unresolved exception class when finding catch block: java.time.format.DateTimeParseException
```

**Expected behavior**
It should be saved correctly.

**Platform**
Amplify Flutter current supports iOS and Android.  This issue is reproducible in (check all that apply):

I just build the app on the Android platform.

On Android 10, It will save correctly. Only occur on Android 7.1

<details>

  <summary>Output of <code>flutter doctor -v</code></summary>
  
  ```[✓] Flutter (Channel stable, 2.5.2, on macOS 11.4 20F71 darwin-x64, locale en-GB)
    • Flutter version 2.5.2 at /Volumes/ForMac/development/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 3595343e20 (7 weeks ago), 2021-09-30 12:58:18 -0700
    • Engine revision 6ac856380f
    • Dart version 2.14.3

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Volumes/ForMac/development/android
    • Platform android-31, build-tools 30.0.3
    • ANDROID_HOME = /Volumes/ForMac/development/android
    • Java binary at: /Volumes/Macintosh HD/Applications/Android
      Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.5.1, Build version 12E507
    • CocoaPods version 1.11.2

[✗] Chrome - develop for the web (Cannot find Chrome executable at /Applications/Google
    Chrome.app/Contents/MacOS/Google Chrome)
    ! Cannot find Chrome. Try setting CHROME_EXECUTABLE to a Chrome executable.

[✓] Android Studio (version 4.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)

[✓] Android Studio (version 2020.3)
    • Android Studio at /Volumes/Macintosh HD/Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)

[✓] VS Code (version 1.62.3)
    • VS Code at /Volumes/Macintosh HD/Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.28.0

[✓] VS Code
    • VS Code at /Volumes/ForMac/Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.28.0

[✓] Connected device (1 available)
    • V2 (mobile) • VB0220CP20718 • android-arm • Android 7.1.1 (API 25)

! Doctor found issues in 1 category.
```

</details>

<details>

  <summary>Dependencies (<code>pubspec.lock</code>)</summary>
  
  ```# Generated by pub
# See https://dart.dev/tools/pub/glossary#lockfile
packages:
  amplify_analytics_pinpoint:
    dependency: "direct main"
    description:
      name: amplify_analytics_pinpoint
      url: "https://pub.dartlang.org"
    source: hosted
    version: "0.2.9"
  amplify_analytics_plugin_interface:
    dependency: transitive
    description:
      name: amplify_analytics_plugin_interface
      url: "https://pub.dartlang.org"
    source: hosted
    version: "0.2.9"
  amplify_api:
    dependency: "direct main"
    description:
      name: amplify_api
      url: "https://pub.dartlang.org"
    source: hosted
    version: "0.2.9"
  amplify_api_plugin_interface:
    dependency: transitive
    description:
      name: amplify_api_plugin_interface
      url: "https://pub.dartlang.org"
    source: hosted
    version: "0.2.9"
  amplify_auth_cognito:
    dependency: "direct main"
    description:
      name: amplify_auth_cognito
      url: "https://pub.dartlang.org"
    source: hosted
    version: "0.2.9"
  amplify_auth_plugin_interface:
    dependency: transitive
    description:
      name: amplify_auth_plugin_interface
      url: "https://pub.dartlang.org"
    source: hosted
    version: "0.2.9"
  amplify_core:
    dependency: transitive
    description:
      name: amplify_core
      url: "https://pub.dartlang.org"
    source: hosted
    version: "0.2.9"
  amplify_datastore:
    dependency: "direct main"
    description:
      name: amplify_datastore
      url: "https://pub.dartlang.org"
    source: hosted
    version: "0.2.9"
  amplify_datastore_plugin_interface:
    dependency: transitive
    description:
      name: amplify_datastore_plugin_interface
      url: "https://pub.dartlang.org"
    source: hosted
    version: "0.2.9"
  amplify_flutter:
    dependency: "direct main"
    description:
      name: amplify_flutter
      url: "https://pub.dartlang.org"
    source: hosted
    version: "0.2.9"
  amplify_storage_plugin_interface:
    dependency: transitive
    description:
      name: amplify_storage_plugin_interface
      url: "https://pub.dartlang.org"
    source: hosted
    version: "0.2.9"
  async:
    dependency: transitive
    description:
      name: async
      url: "https://pub.dartlang.org"
    source: hosted
    version: "2.8.1"
  boolean_selector:
    dependency: transitive
    description:
      name: boolean_selector
      url: "https://pub.dartlang.org"
    source: hosted
    version: "2.1.0"
  characters:
    dependency: transitive
    description:
      name: characters
      url: "https://pub.dartlang.org"
    source: hosted
    version: "1.1.0"
  charcode:
    dependency: transitive
    description:
      name: charcode
      url: "https://pub.dartlang.org"
    source: hosted
    version: "1.3.1"
  clock:
    dependency: transitive
    description:
      name: clock
      url: "https://pub.dartlang.org"
    source: hosted
    version: "1.1.0"
  collection:
    dependency: transitive
    description:
      name: collection
      url: "https://pub.dartlang.org"
    source: hosted
    version: "1.15.0"
  crypto:
    dependency: transitive
    description:
      name: crypto
      url: "https://pub.dartlang.org"
    source: hosted
    version: "3.0.1"
  date_time_format:
    dependency: transitive
    description:
      name: date_time_format
      url: "https://pub.dartlang.org"
    source: hosted
    version: "2.0.1"
  fake_async:
    dependency: transitive
    description:
      name: fake_async
      url: "https://pub.dartlang.org"
    source: hosted
    version: "1.2.0"
  flutter:
    dependency: "direct main"
    description: flutter
    source: sdk
    version: "0.0.0"
  flutter_lints:
    dependency: "direct dev"
    description:
      name: flutter_lints
      url: "https://pub.dartlang.org"
    source: hosted
    version: "1.0.4"
  flutter_localizations:
    dependency: "direct main"
    description: flutter
    source: sdk
    version: "0.0.0"
  flutter_test:
    dependency: "direct dev"
    description: flutter
    source: sdk
    version: "0.0.0"
  intl:
    dependency: transitive
    description:
      name: intl
      url: "https://pub.dartlang.org"
    source: hosted
    version: "0.17.0"
  lints:
    dependency: transitive
    description:
      name: lints
      url: "https://pub.dartlang.org"
    source: hosted
    version: "1.0.1"
  matcher:
    dependency: transitive
    description:
      name: matcher
      url: "https://pub.dartlang.org"
    source: hosted
    version: "0.12.10"
  meta:
    dependency: transitive
    description:
      name: meta
      url: "https://pub.dartlang.org"
    source: hosted
    version: "1.7.0"
  path:
    dependency: transitive
    description:
      name: path
      url: "https://pub.dartlang.org"
    source: hosted
    version: "1.8.0"
  plugin_platform_interface:
    dependency: transitive
    description:
      name: plugin_platform_interface
      url: "https://pub.dartlang.org"
    source: hosted
    version: "2.0.2"
  sky_engine:
    dependency: transitive
    description: flutter
    source: sdk
    version: "0.0.99"
  source_span:
    dependency: transitive
    description:
      name: source_span
      url: "https://pub.dartlang.org"
    source: hosted
    version: "1.8.1"
  stack_trace:
    dependency: transitive
    description:
      name: stack_trace
      url: "https://pub.dartlang.org"
    source: hosted
    version: "1.10.0"
  stream_channel:
    dependency: transitive
    description:
      name: stream_channel
      url: "https://pub.dartlang.org"
    source: hosted
    version: "2.1.0"
  string_scanner:
    dependency: transitive
    description:
      name: string_scanner
      url: "https://pub.dartlang.org"
    source: hosted
    version: "1.1.0"
  term_glyph:
    dependency: transitive
    description:
      name: term_glyph
      url: "https://pub.dartlang.org"
    source: hosted
    version: "1.2.0"
  test_api:
    dependency: transitive
    description:
      name: test_api
      url: "https://pub.dartlang.org"
    source: hosted
    version: "0.4.2"
  typed_data:
    dependency: transitive
    description:
      name: typed_data
      url: "https://pub.dartlang.org"
    source: hosted
    version: "1.3.0"
  uuid:
    dependency: transitive
    description:
      name: uuid
      url: "https://pub.dartlang.org"
    source: hosted
    version: "3.0.5"
  vector_math:
    dependency: transitive
    description:
      name: vector_math
      url: "https://pub.dartlang.org"
    source: hosted
    version: "2.1.0"
sdks:
  dart: ">=2.12.0 <3.0.0"
  flutter: ">=1.20.0"
  ```

</details>

**Smartphone (please complete the following information):**

 - Device: Xiaomi MAX2
 - OS: Android 7.1.1

**Comment by dnys1** [comment]
Thanks for creating this issue, @MasterShu. I'm able to reproduce the exception, and it seems like an issue with compatibility with older Android SDKs. I've created a ticket with the Android team so that they can address it properly.

https://github.com/aws-amplify/amplify-android/issues/1584

**Comment by jamesonwilliams** [comment]
The java 8 time APIs are meant to be resolved at compile time by the Android build tools.

But you have to enable it explicitly in your project. Have you done so?

Please see the configuration options for "core library desugaring" here:

https://developer.android.com/studio/write/java8-support#library-desugaring

**Comment by atomicalsoftwares** [comment]
I'm still facing this issue.

Has any solution or work around been made available?

**Comment by HuiSF** [comment]
Hi @atomicalsoftwares can you try to set you Android App `minSdkVersion` and  `compileSdkVersion` to 26 and see if it works?

It looks like amplify-android uses java.time API which is available only in Android API level 26+

**Comment by HuiSF** [comment]
This fix for the underlying issue existed in amplify-android has been released with amplify-flutter [v0.6.2](https://github.com/aws-amplify/amplify-flutter/releases/tag/v0.6.2).

Closing this issue. Please feel free to follow up or reopen if anything. Thanks.
